frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -0,0 +1,381 @@
+package dockerfile
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/containerd/continuity/fs/fstest"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/frontend/dockerui"
+	gateway "github.com/moby/buildkit/frontend/gateway/client"
+
+	"github.com/moby/buildkit/frontend/subrequests/lint"
+	"github.com/moby/buildkit/util/testutil/integration"
+	"github.com/moby/buildkit/util/testutil/workers"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"github.com/tonistiigi/fsutil"
+)
+
+var lintTests = integration.TestFuncs(
+	testLintRequest,
+	testLintStageName,
+	testLintNoEmptyContinuations,
+)
+
+func testLintRequest(t *testing.T, sb integration.Sandbox) {
+	integration.SkipOnPlatform(t, "windows")
+	workers.CheckFeatureCompat(t, sb, workers.FeatureFrontendOutline)
+	f := getFrontend(t, sb)
+	if _, ok := f.(*clientFrontend); !ok {
+		t.Skip("only test with client frontend")
+	}
+
+	dockerfile := []byte(`
+	FROM scratch as target
+	COPY Dockerfile \
+
+	.
+	ARG inherited=box
+	copy Dockerfile /foo
+
+	FROM scratch AS target2
+	COPY Dockerfile /Dockerfile
+	`)
+
+	dir := integration.Tmpdir(
+		t,
+		fstest.CreateFile("Dockerfile", []byte(dockerfile), 0600),
+	)
+
+	c, err := client.New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	destDir, err := os.MkdirTemp("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(destDir)
+
+	called := false
+	frontend := func(ctx context.Context, c gateway.Client) (*gateway.Result, error) {
+		res, err := c.Solve(ctx, gateway.SolveRequest{
+			FrontendOpt: map[string]string{
+				"frontend.caps": "moby.buildkit.frontend.subrequests",
+				"requestid":     "frontend.lint",
+				"build-arg:BAR": "678",
+				"target":        "target",
+			},
+			Frontend: "dockerfile.v0",
+		})
+		require.NoError(t, err)
+
+		lintResults, err := unmarshalLintResults(res)
```

Comment:
iiuc `checkLinterWarnings` based tests still only check progress stream and this is the only test that calls `unmarshalLintResults`. Can't we have it so that we have two implementations of `checkLinterWarnings`, one based on progress stream and another based on "frontend.lint" . Then all the testcases run in both modes and no duplication is needed.